### ISSUE TITLE
Error: File to import not found or unreadable

### ISSUE THREAD

**Issue by mttdffy** [sass/sassc-ruby#79]
As of 1.12.0 I am seeing errors like this:

```
ActionView::Template::Error: Error: File to import not found or unreadable: /opt/rbenv/versions/2.4.4/lib/ruby/gems/2.4.0/gems/normalize-rails-4.1.1/vendor/assets/stylesheets/normalize-rails/normalize.
        on line 1 of ../../../../../../opt/rbenv/versions/2.4.4/lib/ruby/gems/2.4.0/gems/normalize-rails-4.1.1/vendor/assets/stylesheets/normalize-rails.scss
```

Downgrading to 1.11.4 I no longer saw this issue.

**Comment by tubbo** [comment]
the issue seems to be related to this: https://github.com/sass/libsass/releases/tag/3.5.3

@xzyfer ^

**Comment by yjukaku** [comment]
```
rails generate new test-sassc
cd test-sassc
bundle
#... add sassc gem via gem 'sassc', '1.12.0'
#... add sassc-rails via gem 'sassc-rails'
```
```scss
# in app/assets/stylesheets/application.scss
@import "vend"
```

```css
/* vendor/assets/stylesheets/vend.css */
.something{
  color: red;
}
```

## With `gem 'sassc', '1.11.4'`
```
bundle exec rake assets:precompile
# ... successfully creates public/assets/application.css
```
```css
/* public/assets/application.css */
.something { 
  color: red;
}

```

## With `gem 'sassc', '1.12.0'`

```
bundle exec rake assets:precompile
rake aborted!
SassC::SyntaxError: Error: File to import not found or unreadable: /Users/whatever/test-sassc/vendor/assets/stylesheets/vend.
        on line 17 of app/assets/stylesheets/application.scss
>> @import "vend"
```

**Comment by yjukaku** [comment]
> the issue seems to be related to this: https://github.com/sass/libsass/releases/tag/3.5.3

@tubbo I spent the majority of my morning looking through this and related issues, but what I can't understand is why that release would actually break anything. It seems that all that was added was a deprecation warning (which btw is pending to be rolled back, via https://github.com/sass/libsass/pull/2684). 

@xzyfer do you mind confirming that this release only added a deprecation warning. (also, thanks for all of your patience and work regarding this issue, I've seen how much effort you've put into these libraries!)

**Comment by xzyfer** [comment]
Please file an issue with the gem repo https://github.com/sass/sassc-rails

**Comment by deivid-rodriguez** [comment]
@xzyfer I don't think `sassc-rails` is related. That gem has not seen changes for a long time whereas this gem made a release yesterday. The breaking change seems to be caused indeed by the upgrade in the libsass vendored by this gem.

**Comment by yjukaku** [comment]
`sassc-rails` is definitely not related. Here's an updated example that doesn't use Rails at all, just `bundler` and `sassc`.

## Setup
```
mkdir sassc-imports
cd sassc-imports
bundle init
echo "ruby '2.4.3'" >> Gemfile
echo "2.4.3" >> .ruby-version
echo "gem 'sassc', '1.11.4'" >> Gemfile
bundle
echo ".something{color: red}" > test.css
```

## With SassC 1.11.4
```
bundle exec ruby -e "require 'sassc'; puts SassC::Engine.new(\"@import 'test'\").render"
.something {
  color: red; }
```

## With SassC 1.12.0
```
bundle exec ruby -e "require 'sassc'; puts SassC::Engine.new(\"@import 'test'\").render"
stdin:1: Error: File to import not found or unreadable: test. (SassC::SyntaxError)
        on line 1 of stdin
>> @import 'test'
   ^
	from /Users/whatever/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/gems/sassc-1.12.0/lib/sassc/engine.rb:47:in `render'
```

**Comment by tubbo** [comment]
@xzyfer This is the proper repo for this issue, as it's an issue with the version of libsass that sassc-ruby has vendored in.

The submodule for libsass in this repo points to https://github.com/sass/libsass/commits/1e52b74306b7d73a617396c912ca436dc55fd4d8

But, that commit is _before_ the fix that would have prevented all of these issues. So we really need libsass to be at https://github.com/sass/libsass/tree/19fd2bf2e31413b5935e7feb87fec630b486feba.

**Comment by bolandrm** [comment]
@tubbo i'd prefer not to point at an arbitrary commit, but rather a release...  Is there a release that resolves this?  I don't have much time to dig into this at the moment.

**Comment by deivid-rodriguez** [comment]
It seems to me that `libssass` intends to release a new version in the 3.5 series fixing this, so it should be enough to update the vendored version of libsass once that happens and release 1.12.1 after that.

**Comment by bolandrm** [comment]
Cool, when that is done I'll update to it.

**Comment by bolandrm** [comment]
If anyone is interested in contributing to the project, it would be nice to have a failing test for this.

**Comment by yjukaku** [comment]
@bolandrm Unfortunately I don't think that's really an option. This is a "bug" in libsass, not in this library. And I say "bug", because it's actually something that will be unsupported in the future, but was temporarily rolled back for now. (AFAICT)

**Comment by tubbo** [comment]
Not yet @bolandrm, these commits are at the top of the `3-5-stable` branch, and I suppose they'll eventually be released into v3.5.5 of libsass, but as of now I just took the top of the stable branch to see if that would fix the problem (which it did). I'm also in agreement with you that it would be better to lock it down to a release, not just an arbitrary commit, we just need a release of libsass in order to proceed.

**Comment by bolandrm** [comment]
@yjukaku sorry, i'm not exactly following what you are saying.  What is your proposed resolution?

**Comment by yjukaku** [comment]
Sorry for being unclear. I meant, I don't think it makes much sense to write a failing test for this issue _in this repo_, since it's really just because of a problem in `libsass`.

I do agree that it would be better to wait until a new release of `libsass` rather than an arbitrary commit to release a new version of `sassc`.

**Comment by deivid-rodriguez** [comment]
Well, had the regression test been present in this repo, the `1.12.0` release wouldn't have included this regression. So, no matter where the root problem lies, a regression test seems useful.

**Comment by deivid-rodriguez** [comment]
I opened #81 including a regression test, by the way!

**Comment by jmarkbrooks** [comment]
Any idea when this will be fixed?  or should users just revert to 1.11.4 for now?

**Comment by bolandrm** [comment]
Just released 1.12.1, please give it a try.  Thanks all!
